DllImportGenerator/DllImportGenerator/Marshalling/MarshallingGenerator.cs

```diff
@@ -182,15 +182,15 @@ internal class MarshallingGenerators
                     }
                     return SafeHandle;
 
-                case { ManagedType: IArrayTypeSymbol { IsSZArray: true, ElementType: ITypeSymbol elementType }, MarshallingAttributeInfo: NoMarshallingInfo }:
-                    return CreateArrayMarshaller(info, context, elementType, NoMarshallingInfo.Instance);
+                case { ManagedType: IArrayTypeSymbol { IsSZArray: true, ElementType: ITypeSymbol elementType }, MarshallingAttributeInfo: ArrayMarshallingInfo(MarshallingInfo elementMarshallingInfo) }:
```

Comment:
Let's collapse this case and make a local function. I really dislike this compound statements in switch cases. They are too dense and I fear we don't fully appreciate all the cases we are or aren't supporting.

Reply:
I think removing the `MarshallingAttributeInfo` and handling that in a sub function is sufficient.

Reply:
Personally I prefer putting all of the cases for metadata that we support in the case statement, otherwise we start depending on ordering of case statements. For example, there was a bug where we didn't support putting a MarshalUsing attribute on a `string` parameter since the case statement for "is this a string" didn't check metadata and the "is this any type marked with MarshalUsing" was after the string check. But I can move the metadata check into the `CreateArrayMarshaller` function if you'd prefer.

Reply:
Okay. Let's continue with this pattern then.

>But I can move the metadata check into the CreateArrayMarshaller function if you'd prefer.

Please and thank you.